eng/pipelines/templates/steps/sync-repo-branch.yml

```diff
@@ -0,0 +1,17 @@
+parameters:
+  Repos: []
+  Branch: master
+
+steps:
+
+- ${{ each repoConfig in parameters.Repos }}:
+  - ${{ each repo in repoConfig }}:
+    - script: |
+        git clone --filter=blob:none --branch ${{ coalesce(repo.value.Branch, parameters.Branch) }} https://github.com/azure/${{ repo.key }}
```

Comment:
Can we echo the commands prior to them being run? It would be nice to have something in the output rather than being forced to open the yml to see what's running.

Reply:
Given these are one-line inline scripts DevOps automatically dumps the content to the logs, so we are good for these steps. I will say that adding an echo command to the inline scripts causes issues when running on windows vs linux (one requires quotes the other doesn't allow quotes :(). So to do the echo command I'd have to use pwsh but then we have the stupid stderr issue where git outputs to stderr :(. For now I think as long as these stay simple one-liners then I think we are good.